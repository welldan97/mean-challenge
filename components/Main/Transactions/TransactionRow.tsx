import classNames from 'classnames';
import { memo } from 'react';
import config from '../../../config';
import { Transaction } from '../../../lib/Transaction';

interface Props {
  transaction: Transaction;
}

const TransactionRow = memo(({ transaction }: Props) => {
  if (transaction.type === 'other' && !config.showExtraData) return null;

  return (
    <tr>
      <td
        className={classNames(
          'font-bold',
          transaction.type === 'send' ? 'text-orange-400' : 'text-blue-200',
        )}
      >
        {transaction.type === 'send' && '⇧'}
        {transaction.type === 'receive' && '⇩'}
        {transaction.type === 'other' && '?'}
      </td>
      <td>
        {new Intl.DateTimeFormat('pt-PT', {
          dateStyle: 'short',
          timeStyle: 'short',
        }).format(new Date(transaction.timestamp * 1000))}
      </td>
      <td>{transaction.timestamp}</td>
      <td className="truncate max-w-[100px]">
        {transaction.type === 'send' && transaction.walletAddress}
        {transaction.type === 'receive' && transaction.address}
        {transaction.type === 'other' && transaction.walletAddress}
      </td>
      <td className="truncate max-w-[100px]">
        {transaction.type === 'send' && transaction.address}
        {transaction.type === 'receive' && transaction.walletAddress}
        {transaction.type === 'other' && '-'}
      </td>
      <td>{transaction.type !== 'other' && transaction.amount / 1000000000}</td>
      {config.showExtraData && (
        <>
          <td>{transaction.balanceBefore / 1000000000}</td>
          <td>{transaction.balanceAfter / 1000000000}</td>
        </>
      )}
    </tr>
  );
});

export default TransactionRow;
