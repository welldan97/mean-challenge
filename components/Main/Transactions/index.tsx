import { memo } from 'react';
import { Transaction } from '../../../lib/Transaction';

interface TransactionRowProps {
  transaction: Transaction;
}

const TransactionRow = memo(({ transaction }: TransactionRowProps) => (
  <tr>
    <td>
      {new Intl.DateTimeFormat('pt-PT', {
        dateStyle: 'short',
        timeStyle: 'short',
      }).format(new Date(transaction.timestamp * 1000))}
    </td>
    <td>{transaction.timestamp}</td>
    <td className="truncate max-w-[100px]">{transaction.from}</td>
    <td className="truncate max-w-[100px]">{transaction.to}</td>
    <td>{transaction.amount / 1000000000}</td>
    <td>{transaction.fromBalance}</td>
    <td>{transaction.toBalance}</td>
  </tr>
));

interface Props {
  transactions: Transaction[];
}

const Transactions = memo(({ transactions }: Props) => (
  <div className="w-[800px]">
    <table className="table table-zebra w-full">
      <thead>
        <tr>
          <th>Time</th>
          <th>Timestamp</th>
          <th>From</th>
          <th>To</th>
          <th>Amount</th>
        </tr>
      </thead>
      <tbody>
        {transactions.map(transaction => (
          <TransactionRow
            key={transaction.timestamp}
            transaction={transaction}
          />
        ))}
      </tbody>
    </table>
  </div>
));

export default Transactions;
