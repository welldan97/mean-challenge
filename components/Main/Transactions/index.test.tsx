import { render } from '@testing-library/react';

import transactions from '../../../lib/test/fixtures/transactions';
import Transactions from '.';

describe('Transactions', () => {
  test('render', () => {
    const { container } = render(<Transactions transactions={transactions} />);

    expect(container).toMatchSnapshot();
  });
});
