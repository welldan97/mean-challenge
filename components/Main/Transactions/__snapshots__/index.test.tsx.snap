// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transactions render 1`] = `
<div>
  <div
    class="w-[800px] mb-16"
  >
    <table
      class="table table-zebra w-full"
    >
      <thead>
        <tr>
          <th />
          <th>
            Time
          </th>
          <th>
            Timestamp
          </th>
          <th>
            From
          </th>
          <th>
            To
          </th>
          <th>
            Amount
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td
            class="font-bold text-orange-400"
          >
            ⇧
          </td>
          <td>
            05/10/22, 21:34
          </td>
          <td>
            1665005686
          </td>
          <td
            class="truncate max-w-[100px]"
          >
            F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23
          </td>
          <td
            class="truncate max-w-[100px]"
          >
            B1HWEeZCC6Hy3PF24TUmnW8x23oxQirkKwootk27kQTb
          </td>
          <td>
            1
          </td>
        </tr>
        <tr>
          <td
            class="font-bold text-orange-400"
          >
            ⇧
          </td>
          <td>
            05/10/22, 21:30
          </td>
          <td>
            1665005445
          </td>
          <td
            class="truncate max-w-[100px]"
          >
            F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23
          </td>
          <td
            class="truncate max-w-[100px]"
          >
            B1HWEeZCC6Hy3PF24TUmnW8x23oxQirkKwootk27kQTb
          </td>
          <td>
            1
          </td>
        </tr>
        <tr>
          <td
            class="font-bold text-orange-400"
          >
            ⇧
          </td>
          <td>
            05/10/22, 16:16
          </td>
          <td>
            1664986617
          </td>
          <td
            class="truncate max-w-[100px]"
          >
            F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23
          </td>
          <td
            class="truncate max-w-[100px]"
          >
            B1HWEeZCC6Hy3PF24TUmnW8x23oxQirkKwootk27kQTb
          </td>
          <td>
            0.04
          </td>
        </tr>
        <tr>
          <td
            class="font-bold text-orange-400"
          >
            ⇧
          </td>
          <td>
            05/10/22, 15:09
          </td>
          <td>
            1664982540
          </td>
          <td
            class="truncate max-w-[100px]"
          >
            F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23
          </td>
          <td
            class="truncate max-w-[100px]"
          >
            B1HWEeZCC6Hy3PF24TUmnW8x23oxQirkKwootk27kQTb
          </td>
          <td>
            0.03
          </td>
        </tr>
        <tr>
          <td
            class="font-bold text-orange-400"
          >
            ⇧
          </td>
          <td>
            05/10/22, 15:07
          </td>
          <td>
            1664982449
          </td>
          <td
            class="truncate max-w-[100px]"
          >
            F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23
          </td>
          <td
            class="truncate max-w-[100px]"
          >
            B1HWEeZCC6Hy3PF24TUmnW8x23oxQirkKwootk27kQTb
          </td>
          <td>
            0.03
          </td>
        </tr>
        <tr>
          <td
            class="font-bold text-orange-400"
          >
            ⇧
          </td>
          <td>
            05/10/22, 13:50
          </td>
          <td>
            1664977847
          </td>
          <td
            class="truncate max-w-[100px]"
          >
            F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23
          </td>
          <td
            class="truncate max-w-[100px]"
          >
            B1HWEeZCC6Hy3PF24TUmnW8x23oxQirkKwootk27kQTb
          </td>
          <td>
            0.03
          </td>
        </tr>
        <tr>
          <td
            class="font-bold text-orange-400"
          >
            ⇧
          </td>
          <td>
            05/10/22, 13:33
          </td>
          <td>
            1664976815
          </td>
          <td
            class="truncate max-w-[100px]"
          >
            F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23
          </td>
          <td
            class="truncate max-w-[100px]"
          >
            B1HWEeZCC6Hy3PF24TUmnW8x23oxQirkKwootk27kQTb
          </td>
          <td>
            0.03
          </td>
        </tr>
        <tr>
          <td
            class="font-bold text-orange-400"
          >
            ⇧
          </td>
          <td>
            05/10/22, 12:59
          </td>
          <td>
            1664974742
          </td>
          <td
            class="truncate max-w-[100px]"
          >
            F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23
          </td>
          <td
            class="truncate max-w-[100px]"
          >
            B1HWEeZCC6Hy3PF24TUmnW8x23oxQirkKwootk27kQTb
          </td>
          <td>
            1
          </td>
        </tr>
        <tr>
          <td
            class="font-bold text-blue-200"
          >
            ⇩
          </td>
          <td>
            05/10/22, 00:02
          </td>
          <td>
            1664928170
          </td>
          <td
            class="truncate max-w-[100px]"
          >
            9B5XszUGdMaxCZ7uSQhPzdks5ZQSmWxrmzCSvtJ6Ns6g
          </td>
          <td
            class="truncate max-w-[100px]"
          >
            F4cjCGtoG3Q58c8Qm3DQSq7m74nDYvUBFwSqReRAeZ23
          </td>
          <td>
            2
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
